contracts/helpers/TransferHelper.sol

```diff
@@ -81,113 +71,18 @@ contract TransferHelper is
         TransferHelperItemsWithRecipient[] calldata items,
         bytes32 conduitKey
     ) external override returns (bytes4 magicValue) {
-        // If no conduitKey is given, use TokenTransferrer to perform transfers.
+        // Ensure that a conduit key has been supplied.
         if (conduitKey == bytes32(0)) {
-            _performTransfersWithoutConduit(items);
-        } else {
-            // Otherwise, a conduitKey was provided.
-            _performTransfersWithConduit(items, conduitKey);
+            revert InvalidConduit(conduitKey, address(0));
         }
 
+        // Use conduit derived from supplied conduit key to perform transfers.
+        _performTransfersWithConduit(items, conduitKey);
+
         // Return a magic value indicating that the transfers were performed.
         magicValue = this.bulkTransfer.selector;
     }
 
-    /**
-     * @notice Perform multiple transfers to individually-specified recipients
-     *         via TokenTransferrer.
-     *
-     * @param transfers The transfers to perform.
-     */
-    function _performTransfersWithoutConduit(
-        TransferHelperItemsWithRecipient[] calldata transfers
-    ) internal {
-        // Retrieve total number of transfers and place on stack.
-        uint256 numTransfers = transfers.length;
-
-        // Skip overflow checks: all for loops are indexed starting at zero.
-        unchecked {
-            // Iterate over each transfer.
-            for (uint256 i = 0; i < numTransfers; ++i) {
-                // Retrieve the transfer in question.
-                TransferHelperItemsWithRecipient calldata transfer = transfers[
-                    i
-                ];
-
-                // Retrieve the items of the transfer in question.
-                TransferHelperItem[] calldata transferItems = transfer.items;
-
-                // Ensure recipient is not the zero address.
-                _checkRecipientIsNotZeroAddress(transfer.recipient);
-
-                // Create a boolean indicating whether validateERC721Receiver
-                // is true and recipient is a contract.
-                bool callERC721Receiver = transfer.validateERC721Receiver &&
-                    transfer.recipient.code.length != 0;
```

Comment:
Note that checking code was removed. If you try to check the receiver on an EOA, it won't return magic value.